repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/contract/ContractPreProcessor.java

```diff
@@ -79,14 +79,10 @@ private void processUpdateContract(AtlasEntity entity, EntityMutationContext con
         AtlasEntity existingContractEntity = entityRetriever.toAtlasEntity(vertex);
         // No update to relationships allowed for the existing contract version
         resetAllRelationshipAttributes(entity);
-        DataContract contract = DataContract.deserialize(contractString);
-        String existingContractString = getContractString(existingContractEntity);
-        boolean requestFromMigration = RequestContext.get().getRequestContextHeaders().getOrDefault(
-                "x-atlan-request-id", "").contains("json-to-yaml-migration");
-        if (!requestFromMigration && !StringUtils.isEmpty(contractString) &&
-                !contract.equals(DataContract.deserialize(existingContractString))) {
+        if (entity.getAttribute(ATTR_CERTIFICATE_STATUS) == DataContract.Status.VERIFIED.name() ||
```

Comment:
Why this check done on entity as well?

Reply:
If the entity state is already published, we shouldn't allow any edits to the entity.